app/scenes/DocumentNew.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+import * as React from 'react';
+import { inject } from 'mobx-react';
+import type { RouterHistory, Location } from 'react-router-dom';
+import Flex from 'shared/components/Flex';
+import CenteredContent from 'components/CenteredContent';
+import LoadingPlaceholder from 'scenes/Document/components/LoadingPlaceholder';
+import DocumentsStore from 'stores/DocumentsStore';
+import UiStore from 'stores/UiStore';
+import { documentEditUrl } from 'utils/routeHelpers';
+
+type Props = {
+  history: RouterHistory,
+  location: Location,
+  documents: DocumentsStore,
+  ui: UiStore,
+  match: Object,
+};
+
+class DocumentNew extends React.Component<Props> {
+  async componentDidMount() {
```

Comment:
🍖 When this route is mounted immediately create the new doc and then forward to it.